### PR TITLE
BL-7915 Verify vehicle information before sending notification

### PR DESCRIPTION
**Pull request by michalf-kainos**

### CODE REVIEW

File: motr-notifier/src/main/java/uk/gov/dvsa/motr/notifier/processing/model/SubscriptionQueueItem.java [hunk 1]
```diff
@@ -172,7 +172,7 @@ public String toString() {
                 ", dvlaId='" + dvlaId + '\'' +
                 ", email='" + contactDetail.getValue() + '\'' +
                 ", loadedOnDate=" + loadedOnDate +
-                ", vehicleType=" + vehicleType +
+                ", vehicleType'" + (vehicleType != null ? vehicleType.name() : null) + '\'' +
```
Comment (zakrzewskam) [inline motr-notifier/src/main/java/uk/gov/dvsa/motr/notifier/processing/model/SubscriptionQueueItem.java]: syntax error

File: motr-common/src/main/java/uk/gov/dvsa/motr/vehicledetails/VehicleDetailsClient.java [hunk 2]
```diff
@@ -142,4 +157,9 @@ public VehicleDetailsClient withByRegNumberUri(String uriByRegNumber) {
         this.uriByRegNumber = uriByRegNumber;
         return this;
     }
+
+    public VehicleDetailsClient withHgvPsvByVrmUri(String hgvPsvByVrmUri) {
+        this.uriHgvPsvByVrm = hgvPsvByVrmUri;
```
Comment (zakrzewskam) [inline motr-common/src/main/java/uk/gov/dvsa/motr/vehicledetails/VehicleDetailsClient.java]: Code style of this method is different (missed empty line).

File: motr-notifier/src/integration-test/java/uk/gov/dvsa/motr/test/integration/unloader/SubscriptionDbItemQueueItemUnloaderCommercialVehiclesTest.java [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        }
+
+        SubscriptionDbItem changedSubscriptionDbItem = subscriptionContainer.get();
+
+        // Update vrm in subscriptionItem to match changeSubscriptionDbItem so cleanUp() will find it in the db.
+        subscriptionItem.setVrm(changedSubscriptionDbItem.getVrm());
+        return changedSubscriptionDbItem;
+    }
+
+    private Context buildContext() {
```
Comment (zakrzewskam) [inline motr-notifier/src/integration-test/java/uk/gov/dvsa/motr/test/integration/unloader/SubscriptionDbItemQueueItemUnloaderCommercialVehiclesTest.java]: It should be excluded from this class.

File: motr-notifier/src/main/java/uk/gov/dvsa/motr/notifier/processing/model/SubscriptionQueueItem.java (hunk 1 above)
Comment (michalf-kainos) [reply motr-notifier/src/main/java/uk/gov/dvsa/motr/notifier/processing/model/SubscriptionQueueItem.java]: change removed

File: motr-common/src/main/java/uk/gov/dvsa/motr/vehicledetails/VehicleDetailsClient.java (hunk 2 above)
Comment (michalf-kainos) [reply motr-common/src/main/java/uk/gov/dvsa/motr/vehicledetails/VehicleDetailsClient.java]: aligned

File: motr-notifier/src/integration-test/java/uk/gov/dvsa/motr/test/integration/unloader/SubscriptionDbItemQueueItemUnloaderCommercialVehiclesTest.java (hunk 3 above)
Comment (michalf-kainos) [reply motr-notifier/src/integration-test/java/uk/gov/dvsa/motr/test/integration/unloader/SubscriptionDbItemQueueItemUnloaderCommercialVehiclesTest.java]: extracted with other support methods to abstract parent class

**Pull request merged**